prody/dynamics/adaptive.py

```diff
@@ -146,135 +102,135 @@ def _runStep(structureA, structureB, **kwargs):
+                    .format('%4.3f'%c_sq[0], mode_ids[0]+1))
     else:
-        LOGGER.info('Using {0} modes with cumulative overlap {1} (Modes {2}, ... and {3}) with max mode number {4} and min mode number {5}'
-                    .format(numModes, '{:4.3f}'.format(np.sqrt(c_sq[numModes-1])),
-                            ', '.join([str(entry)
-                                       for entry in modesetA.getIndices()[:10]+1]),
-                            str(modesetA.getIndices()[numModes-1]+1),
-                            np.max(modesetA.getIndices()[:numModes]+1),
-                            np.min(modesetA.getIndices()[:numModes]+1)))
-
-    if np.max(modesetA.getIndices()[:numModes]) > n_modes-5:
+        LOGGER.info('Using {0} modes with square cumulative overlap {1} (Max mode number {2})'
+                    .format(n_sel_modes, '%4.3f'%c_sq[torf_Fmin].max(), np.max(mode_ids)+1))
+
+    if np.max(mode_ids) > n_modes-5:
         n_modes *= 10
 
     if n_modes > dof:
         n_modes = dof
 
-    v = np.sum(np.multiply(overlaps[:numModes], modesetA.getEigvecs()[:, :numModes]),
-               axis=1).reshape(coordsA.shape)
+    v = modes.getEigvecs().dot(overlaps[torf_Fmin])
+    s = f * v.dot(d) / v.dot(v)
+
+    # update coords_init
+    coords_init += s * v.reshape(coords_init.shape)
+    #initial[:] = coords_init[:] # turn this on in case coords_init is not initial in the future
+    rmsd = calcRMSD(coords_init, coords_tar, weights)
+    rmsds.append(rmsd)
 
-    s_min = sum(np.multiply(v.flatten(), d)) / sum(np.power(v.flatten(), 2))
+    if callback_func is not None:
+        cbkwargs = {'coords_init': coords_init, 
+                    'coords_tar': coords_tar, 
+                    'modes': modes, 
+                    'defvec': d}
+        callback_func(**cbkwargs)
 
-    new_coordsA = coordsA + f * s_min * v
+    # deposit 
+    ensemble.addCoordset(coords_init.copy())
+    converged = checkConvergence(rmsds, coords_init, **kwargs)
 
-    rmsd = calcRMSD(new_coordsA, coordsB)
+    if converged:
+        n_modes = 0
 
     LOGGER.info('Current RMSD is {:4.3f}\n'.format(rmsd))
 
-    return rmsd, new_coordsA, modesetA, modesCrossingFmin, dList, n_modes
+    return n_modes
 
-def _checkConvergence(rmsds, coordsA, coordsB, **kwargs):
+def checkConvergence(rmsds, coords, **kwargs):
     """Check convergence of adaptive ANM. 
 
     Convergence is reached if one of three conditions is met:
-    1. difference between *rmsds* from previous step to current < *rmsd_diff_cutoff*
+    1. difference between *rmsds* from previous step to current < *min_rmsd_diff*
     2. Current rmsd < *target_rmsd*
-    3. A node in *coordsA* or *coordsB* gets disconnected from another by > *cutoff*
+    3. A node in *coords* gets disconnected from another by > *cutoff*
 
     :arg rmsds: a list of RMSDs from Adaptive ANM
     :type rmsds: list
 
     :arg coordsA: coordinate set A for checking disconnections
```

Comment:
this should be coords not coordsA and the description needs to change accordingly